src/ResourceControllers/CustomMenu.php

```
@@ -0,0 +1,124 @@
+<?php
+
+// phpcs:disable PSR1.Methods.CamelCapsMethodName
+
+namespace WpYaml\ResourceControllers;
+
+/**
+ * @SuppressWarnings(PHPMD.CamelCaseVariableName)
+ * @SuppressWarnings(PHPMD.CamelCaseMethodName)
+ */
+class CustomMenu extends ResourceController
+{
+    public function process()
+    {
+        if (!empty($this->config)) {
+            add_action('admin_menu', [$this, 'loadMenus']);
+        }
+    }
+
+    /**
+     * Function loadSubMenu
+     *
+     * @return void
+     */
+    public function loadMenus()
+    {
+        foreach ($this->config as $menu) {
+            if (!isset($menu['capability'])) {
+                throw new \Exception('capability is required for ' . $menu['type']);
+            }
+
+            if ($menu['type'] === 'main_menu') {
```

How do you feel about `switch` instead of 2 `if`s? 

We could use the `default` case to throw an exception that `$menu['type']` contains an unsupported value this way